Sources/MapboxNavigation/Match.swift

```diff
@@ -14,19 +14,18 @@ extension Match {
      - returns: A polyline whose length is twice `distance` and whose centroid is located at the maneuver.
      */
     func polylineAroundManeuver(legIndex: Int, stepIndex: Int, distance: CLLocationDistance) -> LineString {
-        let precedingLegs = legs.prefix(upTo: legIndex)
-        let precedingLegCoordinates = precedingLegs.flatMap { $0.steps }.flatMap { $0.shape?.coordinates ?? [] }
-        
-        let precedingSteps = legs[legIndex].steps.prefix(upTo: stepIndex)
-        let precedingStepCoordinates = precedingSteps.compactMap { $0.shape?.coordinates }.reduce([], +)
-        let precedingPolyline = LineString((precedingLegCoordinates + precedingStepCoordinates).reversed())
+        var precedingCoordinates = [LocationCoordinate2D]()
+        if stepIndex > 0 {
+            precedingCoordinates = legs[legIndex].steps[safe: stepIndex - 1]?.shape?.coordinates ?? []
+        }
+        if precedingCoordinates.isEmpty, legIndex > 0 {
+            precedingCoordinates = legs[legIndex - 1].steps.suffix(2).flatMap { $0.shape?.coordinates ?? [] }
+        }
```

Comment:
I’m curious about how the new code will differ in behavior from the old code. If I remember correctly, the old code could include coordinates from any nearby step or leg, since step and leg boundaries are irrelevant to arrow-drawing. If there happen to be multiple intermediate waypoints close together, the new code would crop the arrow at just the adjacent step on either side of the intersection. In exchange, the code is probably a little more performant because it doesn’t have to reverse the arrow of coordinates. This might be a reasonable tradeoff for performance, but introducing `lazy` would probably have the same effect.

The new code appears to be [consistent with Android](https://github.com/mapbox/mapbox-navigation-android/blob/f528c3011438ba31c3383382697129fd9f581f74/libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/arrow/RouteArrowUtils.kt#L27-L39). However, I’d contend that the current behavior on Android is technically a bug, not a feature. It just isn’t a severe enough bug to go out of our way to fix it, but conversely I’m not seeing a strong case for going out of our way to introducing the same behavior on iOS.

Reply:
Thanks for the suggestion, the new code is reverted to keep the previous function to extend the arrow to any nearby steps or legs.